modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/WindowStage.java

```diff
@@ -832,7 +832,10 @@ public void closePostponed() {
     // closed notification. This state is necessary to prevent the platform
     // window from being closed more than once.
     void setPlatformWindowClosed() {
-        platformWindow = null;
+        if (platformWindow != null) {
+            platformWindows.remove(platformWindow);
+            platformWindow = null;
+        }
```

Comment:
This looks like a correct fix. If this method is ever called with a non-null platformWindow, we can be assured that a leak will occur. Can you see whether it would be possible to create an automated test? It would require triggering a WindowEvent.DESTROY before the close method is called.